src/pkg/cluster/secrets.go

```diff
@@ -138,11 +117,8 @@ func (c *Cluster) UpdateZarfManagedImageSecrets(ctx context.Context, state *type
 		if err != nil {
 			return err
 		}
-		if maps.EqualFunc(currentRegistrySecret.Data, newRegistrySecret.Data, func(v1, v2 []byte) bool { return bytes.Equal(v1, v2) }) {
```

Comment:
Given that apply operations do not cost much, even if the data is the same, IMO it's better to simplify the code and let the k8s server figure out what to change / keep the same.